File: java/kode program/Makanan.java
// Makanan.java
public class Makanan extends Menu {
    private String pedas;
    private String jenisMasakan;

    public Makanan(String nama, int harga, String kodeMenu, String deskripsi, String pedas, String jenisMasakan) {
        // Memanggil constructor dari superclass (Menu)
        super(nama, harga, kodeMenu, deskripsi);
        this.pedas = pedas;
        this.jenisMasakan = jenisMasakan;
    }

    // Getter untuk atribut spesifik Makanan
    public String getPedas() { return pedas; }
    public String getJenisMasakan() { return jenisMasakan; }

    @Override
    public void tampilkanDetail() {
        System.out.printf("[%s] %s - Rp%,d\n", this.kodeMenu, this.nama, this.harga);
        System.out.printf("    Deskripsi: %s\n", this.deskripsi);
        System.out.printf("    Info: Pedas level %s, Jenis: %s\n", this.pedas, this.jenisMasakan);
    }
}

File: python/kode program/makanan.py
# makanan.py

from menu import Menu

class Makanan(Menu):
    def __init__(self, nama, harga, kode_menu, deskripsi, pedas, jenis_masakan):
        super().__init__(nama, harga, kode_menu, deskripsi)
        self.pedas = pedas  # Tingkat kepedasan
        self.jenis_masakan = jenis_masakan

    def tampilkan(self):
        # Memanggil metode tampilkan dari parent class dan menambahkan info spesifik
        super().tampilkan()
        print(f"    Info: Pedas level {self.pedas}, Jenis: {self.jenis_masakan}")